### PR TITLE
Substitute hardcoded string encoding with Java Charset constants

### PR DESCRIPTION
**Pull request by bhattacharyyasom**
This change will make the code safer from unsupported encoding errors due to
programmer mistakes.

### CODE REVIEW

**Comment by bhattacharyyasom** [comment]
Something is going wrong after the `jdk_switcher use oraclejdk9` step causing the check on jdk9 to fail. . Can someone help me understand this ?

**Comment by bhattacharyyasom** [comment]
This build will pass once the fix for jdk 9 build goes in.

**Comment by xuwei-k** [comment]
fixed jdk9 build https://github.com/tototoshi/scala-csv/commit/8afec91fb685e6d81876827ed44bd72a45a411c7

**Comment by bhattacharyyasom** [comment]
The build now passes.

File: src/main/scala/com/github/tototoshi/csv/CSVReader.scala [hunk 1]
```diff
@@ -123,7 +123,7 @@ object CSVReader {
     open(file, this.DEFAULT_ENCODING)(format)
   }
 
-  def open(file: File, encoding: String)(implicit format: CSVFormat): CSVReader = {
```
Comment (tototoshi) [inline src/main/scala/com/github/tototoshi/csv/CSVReader.scala]: For compatibility, please add new methods with `Charset` instead of replacing this.

Comment (bhattacharyyasom) [inline src/main/scala/com/github/tototoshi/csv/CSVReader.scala]: i have added new methods. Please review.

File: src/main/scala/com/github/tototoshi/csv/CSVReader.scala [hunk 2]
```diff
@@ -132,10 +132,21 @@ object CSVReader {
     }
   }
 
+  def open(file: File, encoding: String)(implicit format: CSVFormat): CSVReader = {
```
Comment (tototoshi) [inline src/main/scala/com/github/tototoshi/csv/CSVReader.scala]: About implementation, I think it's better to reuse `open(file: File, encoding: Charset)`

```scala
open(file, Charset.forName(encoding))
```

File: src/main/scala/com/github/tototoshi/csv/CSVWriter.scala [hunk 3]
```diff
@@ -117,29 +118,29 @@ class CSVWriter(protected val writer: Writer)(implicit val format: CSVFormat) ex
 
 object CSVWriter {
 
-  def open(file: File)(implicit format: CSVFormat): CSVWriter = open(file, false, "UTF-8")(format)
+  def open(file: File)(implicit format: CSVFormat): CSVWriter = open(file, false, StandardCharsets.UTF_8)(format)
 
-  def open(file: File, encoding: String)(implicit format: CSVFormat): CSVWriter = open(file, false, encoding)(format)
+  def open(file: File, encoding: Charset)(implicit format: CSVFormat): CSVWriter = open(file, false, encoding)(format)
```
Comment (tototoshi) [inline src/main/scala/com/github/tototoshi/csv/CSVWriter.scala]: Could you do the same thing for CSVWriter too?

Comment (bhattacharyyasom) [inline src/main/scala/com/github/tototoshi/csv/CSVWriter.scala]: Yes i have added the same here. Please let me know if there is anything else.

File: src/main/scala/com/github/tototoshi/csv/CSVReader.scala (hunk 2 above)
Comment (bhattacharyyasom) [inline src/main/scala/com/github/tototoshi/csv/CSVReader.scala]: Done in next commit.

File: src/main/scala/com/github/tototoshi/csv/CSVWriter.scala (hunk 3 above)
Comment (bhattacharyyasom) [inline src/main/scala/com/github/tototoshi/csv/CSVWriter.scala]: Done in next commit.

**Comment by bhattacharyyasom** [comment]
Addressed comments.

**Comment by tototoshi** [comment]
Sorry for the review too late.
I reviewed again. It is almost good, but there are places I want to fix slightly if I consider backward compatibility. It is not a very complicated library and I want to consider compatibility as much as possible.
I'd like to merge it after putting some modifications.

**Comment by bhattacharyyasom** [comment]
I understand. Cool. Thanks. I am learning Scala so trying to learn through looking at code and suggest changes. Thanks.

**Pull request closed without merging**